pkg/apis/monitoring/v1/thanos_types.go

```diff
@@ -214,6 +214,14 @@ type ThanosRulerSpec struct {
 	// +listType=map
 	// +listMapKey=ip
 	HostAliases []HostAlias `json:"hostAliases,omitempty"`
+	// RemoteWriteConfig configures stateless mode in ThanosRuler.
+	// RemoteWriteConfig configurations must have the form as specified in the official Prometheus documentation:
+	// https://thanos.io/tip/components/rule.md/#stateless-ruler-via-remote-write
+	// +optional
+	RemoteWriteConfig []RemoteWriteSpec `json:"remoteWriteConfigFile,omitempty"`
+	// RemoteWriteConfigFile specifies the path of the remote write file.
+	// When used alongside with RemoteWriteConfig, remoteWriteConfigFile takes precedence.
+	RemoteWriteConfigFile *string `json:"remoteWriteConfigFile,omitempty"`
```

Comment:
I'm not sure we need RemoteWriteConfigFile since RemoteWriteConfig brings more validation.

Reply:
I have remove RemoteWriteConfigFile field.